core/eolearn/core/core_tasks.py

```diff
@@ -63,19 +63,20 @@ def __init__(self, path, filesystem=None, create=False, config=None):
         :type config: SHConfig or None
         """
         self.path = path
-        self._filesystem = filesystem
-        self._create = create
-        self.config = config
+        self.filesystem_path = "/" if filesystem is None else self.path
 
-        self.filesystem_path = "/" if self._filesystem is None else self.path
+        self._pickled_filesystem = None if filesystem is None else pickle_fs(filesystem)
+        self._create_path = create
+        self.config = config
 
     @property
     def filesystem(self):
-        """A filesystem property that either initializes a new object or returns an existing one."""
-        if self._filesystem is None:
-            return get_filesystem(self.path, create=self._create, config=self.config)
+        """A filesystem property that unpickles an existing filesystem definition or creates a new one."""
+        if self._pickled_filesystem is None:
+            filesystem = get_filesystem(self.path, create=self._create_path, config=self.config)
+            self._pickled_filesystem = pickle_fs(filesystem)
 
-        return self._filesystem
+        return unpickle_fs(self._pickled_filesystem)
```

Comment:
Is unpickling the fs costly? if yes, then you could add `return filesystem` under the if-statement to reduce unnecessary unpickling operations.

Reply:
My benchmarks show `6 µs` which is nothing. But I still decided to add the return statement.